File: test/IntegrationTests/SalesOrderMockDb.cs

using DevZest.Samples.AdventureWorksLT;

namespace DevZest.Data
{
    /// <remarks><see cref="SalesOrder"/> and <see cref="SalesOrderDetail"/> are chosen for having foreing key to non-existing table(s) and
    /// parent-child relationship.</remarks>
    internal sealed class SalesOrderMockDb : MockDb<Db>
    {
        private static DataSet<SalesOrderHeader> Headers()
        {
            return DataSet<SalesOrderHeader>.ParseJson(Strings.Mock_SalesOrderHeader);
        }

        private static DataSet<SalesOrderDetail> Details()
        {
            return DataSet<SalesOrderDetail>.ParseJson(Strings.Mock_SalesOrderDetail);
        }

        protected override void Initialize()
        {
            // The order of mocking table does not matter, the dependencies will be sorted out automatically.
            Mock(Db.SalesOrderDetail, Details);
            Mock(Db.SalesOrderHeader, Headers);
        }
    }
}

File: samples/AdventureWorksLT.DbDesign.VB/SalesOrderMockDb.vb
Imports DevZest.Data

Public Class SalesOrderMockDb
    Inherits MockDb(Of Db)

    Private Shared Function Headers() As DataSet(Of SalesOrderHeader)
        Return DataSet(Of SalesOrderHeader).ParseJson(My.Strings.Mock_SalesOrderHeader)
    End Function

    Private Shared Function Details() As DataSet(Of SalesOrderDetail)
        Return DataSet(Of SalesOrderDetail).ParseJson(My.Strings.Mock_SalesOrderDetail)
    End Function

    Protected Overrides Sub Initialize()
        Mock(Db.SalesOrderHeader, Function() Headers())
        Mock(Db.SalesOrderDetail, Function() Details())
    End Sub
End Class
